apiserver/facades/client/application/deployrepository.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/rpc/params"
+)
+
+// DeployFromRepository is a one-stop deployment method for repository
+// charms. Only a charm name is required to deploy. If argument validation
+// fails, a list of all errors found in validation will be returned. If a
+// local resource is provided, details required for uploading the validated
+// resource will be returned.
+func (api *APIBase) DeployFromRepository(args params.DeployFromRepositoryArgs) (params.DeployFromRepositoryResults, error) {
+	if !featureflag.Enabled(feature.ServerSideCharmDeploy) {
+		return params.DeployFromRepositoryResults{}, errors.NotImplementedf("this facade method is under develop")
+	}
+
+	if err := api.checkCanWrite(); err != nil {
+		return params.DeployFromRepositoryResults{}, errors.Trace(err)
+	}
+	if err := api.check.RemoveAllowed(); err != nil {
+		return params.DeployFromRepositoryResults{}, errors.Trace(err)
+	}
+
+	results := make([]params.DeployFromRepositoryResult, len(args.Args))
+	for i, entity := range args.Args {
+		info, pending, errs := api.deployOneFromRepository(entity)
+		if len(errs) > 0 {
+			results[i].Errors = apiservererrors.ServerErrors(errs)
+			continue
+		}
+		results[i].Info = info
+		results[i].PendingResourceUploads = pending
+	}
+	return params.DeployFromRepositoryResults{
+		Results: results,
+	}, nil
+}
+
+func (api *APIBase) deployOneFromRepository(arg params.DeployFromRepositoryArg) ([]string, []*params.PendingResourceUpload, []error) {
```

Comment:
Are we going to have a `deployTwoFromRepository` function? Maybe we can have a better naming.

Reply:
Adding "one" to the method name is a convention when a facade call can handle more than one of a thing. It splits the business logic of what is happening from the logic of processing more than one item.